ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/ExecutionEngineJsonRpcMethod.java

```diff
@@ -106,4 +123,22 @@ public final JsonRpcResponse response(final JsonRpcRequestContext request) {
   }
 
   public abstract JsonRpcResponse syncResponse(final JsonRpcRequestContext request);
+
+  protected ValidationResult<RpcErrorType> validateForkSupported(final long blockTimestamp) {
+    if (protocolSchedule.isPresent()) {
+      var cancun =
+          protocolSchedule.get().hardforkFor(s -> s.fork().name().equalsIgnoreCase("Cancun"));
```

Comment:
This could be done once in the  constructor.

Reply:
done